lib/internal/test_runner/reporter/junit.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+const {
+  ArrayPrototypeFilter,
+  ArrayPrototypeMap,
+  ArrayPrototypeJoin,
+  ArrayPrototypePush,
+  ArrayPrototypeSome,
+  NumberPrototypeToFixed,
+  ObjectEntries,
+  RegExpPrototypeSymbolReplace,
+  String,
+  StringPrototypeRepeat,
+} = primordials;
+
+const { inspectWithNoCustomRetry } = require('internal/errors');
+const { hostname } = require('os');
+
+const inspectOptions = { __proto__: null, colors: false, breakLength: Infinity };
+const HOSTNAME = hostname();
+
+function escapeProperty(s = '') {
+  return escapeContent(RegExpPrototypeSymbolReplace(/"/g, RegExpPrototypeSymbolReplace(/\n/g, s, ''), '\\"'));
+}
+
+function escapeContent(s = '') {
+  return RegExpPrototypeSymbolReplace(/</g, RegExpPrototypeSymbolReplace(/>/g, s, '&gt;'), '&lt;');
```

Comment:
I think we don't actually need to escape ">" and also need to escape & for this to be correct, quoting the spec:

> The ampersand character (&) and the left angle bracket (<) must not appear in their literal form, except when used as markup delimiters, or within a [comment](https://www.w3.org/TR/xml/#dt-comment), a [processing instruction](https://www.w3.org/TR/xml/#dt-pi), or a [CDATA section](https://www.w3.org/TR/xml/#dt-cdsection). If they are needed elsewhere, they must be [escaped](https://www.w3.org/TR/xml/#dt-escape) using either [numeric character references](https://www.w3.org/TR/xml/#dt-charref) or the strings " &amp; " and " &lt; " respectively. The right angle bracket (>) may be represented using the string " &gt; ", and must, [for compatibility](https://www.w3.org/TR/xml/#dt-compat), be escaped using either " &gt; " or a character reference when it appears in the string " ]]> " in content, when that string is not marking the end of a [CDATA section](https://www.w3.org/TR/xml/#dt-cdsection).

Reply:
> The ampersand character (&) and the left angle bracket (<) must not appear in their literal form, except when used as markup delimiters, or within a [comment](https://www.w3.org/TR/xml/#dt-comment), a [processing instruction](https://www.w3.org/TR/xml/#dt-pi), or a [CDATA section](https://www.w3.org/TR/xml/#dt-cdsection). If they are needed elsewhere, they must be [escaped](https://www.w3.org/TR/xml/#dt-escape) using either [numeric character references](https://www.w3.org/TR/xml/#dt-charref) or the strings " &amp; " and " &lt; " respectively. The right angle bracket (>) may be represented using the string " &gt; ", and must, [for compatibility](https://www.w3.org/TR/xml/#dt-compat), be escaped using either " &gt; " or a character reference when it appears in the string " ]]> " in content, when that string is not marking the end of a [CDATA section](https://www.w3.org/TR/xml/#dt-cdsection).

Reply:
markup unescapes this :/ it seems &lt and &gt should be used:
https://www.w3.org/TR/xml/#syntax